### PR TITLE
Add e2e measure tool for API coverage

### LINKED ISSUE #5701: Measure kubernetes API test coverage on e2e test
According to the official document of the Kubernetes API[1], APIs are classified to three levels: alpha, beta and stable. Alpha and beta APIs can be changed and it is not mandatory to verify these APIs. On the other hand, we want to test the stable APIs as possible to know every single PRs doesn’t affect these APIs’ behavior for the stability. In addition, we want to know which APIs are called at conformance tests because companies which hope to obtain Certified Kubernetes mark need to provide these APIs as services.
We have some discussion e2e test should cover all APIs or not. At least, we need to know which APIs are tested before deciding we should do that.

The corresponding design documentation is [2]

This issue is a part of https://github.com/kubernetes/test-infra/issues/5231

[1]: https://kubernetes.io/docs/concepts/overview/kubernetes-api/
[2]: https://docs.google.com/document/d/1Upo8lTeR7gy4olD6hASTCdn8Vy7WuDwkO_o9Y5nrxhs/edit#

### PR DESCRIPTION
**Pull request by oomichi**
This patch provides a way to know which APIs of k8s are tested on e2e.
This adds the tool compares the log and Swagger spec of k8s, then
outputs which APIs are not tested.

Usage:
 $ go run hack/e2e.go -- -v --test \
   --test_args="--ginkgo.focus=[Conformance] -v=8" > api.log
 $ go run apicoverage.go --restlog=api.log

Signed-off-by: Nikita Komarov <nikita@kmrov.ru>

The design paper is https://docs.google.com/document/d/1Upo8lTeR7gy4olD6hASTCdn8Vy7WuDwkO_o9Y5nrxhs/edit

fixes #5701

### CODE REVIEW

**Comment by BenTheElder** [comment]
sorry, things have been hectic with jenkins and other failures and some others have been off at kubecon
/ok-to-test
/cc @fejta

File: experiment/coverage/apicoverage.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/go-openapi/spec"
+	"github.com/golang/glog"
+)
+
+var (
+	openAPIFile = flag.String("openapi", "https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json", "URL to openapi-spec of Kubernetes. If not specifying, the openapi-spec is download from https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json instead")
```
Comment (fejta) [inline experiment/coverage/apicoverage.go]: Does -h print the default value? Seems overly long

File: experiment/coverage/apicoverage.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	var found bool
+	var apisTested apiArray
+	var apisNotTested apiArray
+	var apisNotTestedAlpha apiArray
+	var apisNotTestedBeta apiArray
+	var apisNotTestedStable apiArray
+
+	flag.Parse()
+	if len(*restLog) == 0 {
+		glog.Fatal("need to set '--restlog'")
```
Comment (fejta) [inline experiment/coverage/apicoverage.go]: IMO the '' part isn't necessary

File: experiment/coverage/apicoverage.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/go-openapi/spec"
+	"github.com/golang/glog"
+)
+
+var (
```
Comment (fejta) [inline experiment/coverage/apicoverage.go]: A useful future improvement might be a `--minimumCoverage=0.5` or something, and then at the end of the main function fail if the number of apis coverered is less than 50% 

Then we can dump this into a regression test

**Comment by fejta** [comment]
Feel free to `/hold cancel` when you're ready for this to go in!

File: experiment/coverage/apicoverage.go (hunk 1 above)
Comment (oomichi) [reply experiment/coverage/apicoverage.go]: oh, nice point. Current help is following and that is very long and redundant. I will remove the part of (default ... ) at least.

  -openapi string
        URL to openapi-spec of Kubernetes. If not specifying, the openapi-spec is download from https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json instead (default "https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json")

File: experiment/coverage/apicoverage.go (hunk 3 above)
Comment (oomichi) [reply experiment/coverage/apicoverage.go]: Cool, that might be useful for operating this tool on CI system to notify warning of coverage.
I think we will have some points we want to discuss related to this feature.
So I will register this point as an issue after first merger.

Comment (oomichi) [reply experiment/coverage/apicoverage.go]: This is registered as https://github.com/kubernetes/test-infra/issues/6079

**Pull request merged**